core/src/main/java/com/bankbazaar/tripmanager/manager/TripActivityManager.java

```diff
@@ -26,24 +26,34 @@ public Optional<TripActivity> getTripActivityById(Long id) {
     }
     /**
      * Delete record by id
+     *
      * @param id
      */
-    public String deleteTripActivity(Long id) {
-        if(tripActivityRepository.findById(id).isEmpty())
+    public boolean deleteTripActivity(Long id) {
+        if(CheckData(id).isPresent())
         {
-            return null;
+            tripActivityRepository.deleteById(id);
+            return true;
         }
-        tripActivityRepository.deleteById(id);
-        return "TripActivity "+id+" has been removed" ;
+        return false;
     }
     /**
      * Update record
      */
     public TripActivity updateTripActivity(TripActivity tripActivity) {
-        if(tripActivityRepository.findById(tripActivity.getTripId()).isEmpty())
+
+        TripActivity presentData = CheckData(tripActivity.getTripId()).orElse(null);
+        if(presentData!=null)
```

Comment:
When we are using optional try to not due null comparisons
Optional<TripActivity> tripOptional = = CheckData(tripActivity.getTripId());
if (tripOptional.isPresent()){
...
}

Reply:
Missed out on some parts. i have made the corrections